src/riak_ql_parser.yrl

```diff
@@ -416,30 +424,80 @@ convert(#outputs{type    = select,
                  buckets = B,
                  fields  = F,
                  limit   = L,
-                 where   = W}) ->
+                 where   = W,
+                 group_by = G} = Outputs) ->
+    ok = validate_select_query(Outputs),
     [
      {type, select},
      {tables, B},
      {fields, F},
      {limit, L},
-     {where, W}
+     {where, W},
+     {group_by, G}
     ];
 convert(#outputs{type = create} = O) ->
     O.
 
-%% make_atom({binary, SomeWord}) ->
-%%     {atom, binary_to_atom(SomeWord, utf8)}.
+validate_select_query(Outputs) ->
+    ok = assert_group_by_select(Outputs).
+
+%% If the query uses GROUP BY then check that the identifiers in the select
+%% all exist in the GROUP BY.
+assert_group_by_select(#outputs{ group_by = [] }) ->
+    ok;
+assert_group_by_select(#outputs{ fields = Fields, group_by = GroupBy }) ->
```

Comment:
`*` is a valid field identifier for GROUP BY's but this code throws a `Field(s) * are specified in the select statement but not the GROUP BY.` error